source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,443 @@
+business domain standpoint.
+
+It is important to name things with a language shared by the team and prevent
+exposing implementation details (ids, different names…) as much as possible. We
+recommend the reading of the GraphQL documentation page
+[Thinking in Graphs](https://graphql.github.io/learn/thinking-in-graphs/).
+
+Let’s add the code to expose a counter field and a mutation in our graph.
+
+### Define the schema
+
+Front-Commerce contains all the tooling to allow you to describe your schema
+using the expressive
+[GraphQL Schema Definition Language (SDL)](https://graphql.org/learn/schema/#type-language).
+
+Create a `my-module/server/modules/clicks-counters/schema.gql` file to contain
+the schema matching our use case.
+
+We would like to:
+
+1. add a `clicksCounter` field to the existing `Product` type
+2. add an `incrementProductCounter` mutation
+
+It could look like so:
+
+```graphql
+# my-module/server/modules/clicks-counters/schema.gql
+extend type Product {
+  clicksCounter: Int
+}
+
+extend type Mutation {
+  incrementProductCounter(sku: String!, incrementValue: Int): MutationSuccess
+}
+```
+
+<blockquote class="info">
+  The `Product` type is part of Front-Commerce’s `Magento2` module. The
+  `MutationSuccess` type used as the mutation response type is part of
+  Front-Commerce’s core.
+</blockquote>
+
+Then expose the schema using the `typeDefs` key of the module definition. Update
+the `my-module/server/modules/clicks-counters/index.js` entrypoint:
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
++import typeDefs from "./schema.gql";
++
+export default {
+-  namespace: "ClicksCounters"
++  namespace: "ClicksCounters",
++  typeDefs: typeDefs
+};
+```
+
+<blockquote class="info">
+  Webpack is in fact taking care of converting SDL into a Javascript object
+  using an appropriate loader.
+</blockquote>
+
+**Congratulations again!** You should now be able to see these new fields and
+sue them in GraphQL.
+
+Try to execute the query below in your GraphQL playground (by default at
+[http://localhost:4000/playground](http://localhost:4000/playground)):
+
+```graphql
+# http://localhost:4000/playground
+{
+  product(sku: "WH09") {
+    sku
+    clicksCounter
+  }
+}
+```
+
+Our GraphQL module does not yet have any code for sending content, so you must
+see the following response:
+
+```json
+{
+  "data": {
+    "product": {
+      "sku": "WH09",
+      "clicksCounter": null
+    }
+  }
+}
+```
+
+Let’s now add some executable logic to our module.
+
+### Implement resolvers
+
+The GraphQL middleware relies on resolver functions to determine the data to
+return for a given field. This is where most of the « real work » is done, for
+instance by fetching remote datasources or transforming data.
+
+Resolvers are exposed using the `resolvers` key of the module definition. It
+should be a **Resolver map**: an object where each key is a GraphQL type name,
```

Comment:
I'd put a link on to `https://www.apollographql.com/docs/graphql-tools/resolvers` here too.